src/main/java/org/generationcp/middleware/dao/oms/CVTermRelationshipDao.java

```diff
@@ -270,6 +273,22 @@ public Integer retrieveAnalysisDerivedVariableID(final Integer originalVariableI
 		}
 	}
 
+	public List<String> getScaleCategoriesUsedInAttributes(final int scaleId) {
+		final SQLQuery query = this.getSession().createSQLQuery(
+			"SELECT v.name category "
+				+ " FROM cvterm_relationship scale_values "
+				+ " INNER JOIN cvterm v ON v.cvterm_id = scale_values.object_id "
+				+ " WHERE scale_values.subject_id = :scaleId AND scale_values.type_id = " + TermId.HAS_VALUE.getId()
```

Comment:
Better to link variable checking in query:
```
 INNER JOIN cvterm_relationship var ON var.object_id = scale_values.subject_id and var.type_id = " + TermId.HAS_SCALE
				.getId()
```
then in the EXISTS clause:
`WHERE a.cval_id = v.cvterm_id and a.atype = var.subject_id`

Reply:
done.